src/test/scala/com/github/saurfang/sas/SasRelationSpec.scala

```diff
@@ -214,4 +214,42 @@ class SasRelationSpec extends FlatSpec with Matchers with SharedSparkContext {
         row1.toSeq should ===(row2.toSeq)
     }
   }
+
+  "SASReltion" should "read compressed numeric SAS data exactly correct" in {
+
+    val sqlContext = new SQLContext(sc)
+    implicit val dblEquality = tolerantDoubleEquality(ParsoWrapper.EPSILON)
+
+    // Set configs to cause multiple partitions/splits.
+    val conf = sqlContext.sparkContext.hadoopConfiguration
+    conf.setLong("mapred.max.split.size", 2000000)
```

Comment:
Compressed data cannot be split, there is no point in setting "mapred.max.split.size".

Reply:
Some compression algorithms are splittable 

https://www.cloudera.com/documentation/enterprise/5-16-x/topics/admin_data_compression_performance.html

```
if you need your compressed data to be splittable, BZip2 and LZO formats can be split
```

Reply:
If we want to support splitting compressed files, we need to allow it in here:
https://github.com/saurfang/spark-sas7bdat/blob/v2.1.0/src/main/scala/com/github/saurfang/sas/mapreduce/SasInputFormat.scala

And modify the offset calculations in: SasRecordReader.scala